### PR TITLE
Enum rework

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @sdroege @EPashkin

### CODE REVIEW

**Comment by EPashkin** [comment]
Looks good, but `cairo_content_t` seems seems still needed (or `CairoContent` renamed back)
In this way cairo-sys don't need glib dependency in Cargo.toml?

**Comment by GuillaumeGomez** [comment]
I've put back all `*_t` aliases.

**Comment by EPashkin** [comment]
Maybe leave enums in sys with values otherwise sys crate will be less usable?

**Comment by GuillaumeGomez** [comment]
I prefer to move enums outside of it and just keep aliases. It's more consistent with other gtk-rs crates this way.

**Comment by GuillaumeGomez** [comment]
(And for failing crates, I'll have to update the corresponding parts by hand.)

**Comment by EPashkin** [comment]
Not all other gtk-rs crates have enums both in sys and normal.

**Comment by GuillaumeGomez** [comment]
True but the missing `__Unknown` field really bothered me. What do you think of it @sdroege?

**Comment by EPashkin** [comment]
Usually we have `#repr[C] enum` in sys and rust enum with `__Unknown` in main crate.

**Comment by EPashkin** [comment]
.. as you already created rust enums to main you can just restore enums in sys and IMHO it will be good.

**Comment by GuillaumeGomez** [comment]
Which crate has enums in sys? Didn't find one yet (looked into gtk and gdk only for the moment...)

**Comment by EPashkin** [comment]
@GuillaumeGomez Yes, I was wrong.
We use `c_int` type with constants instead enum in sys.

**Comment by EPashkin** [comment]
.. but do it in cairo-rs-sys will be IMHO too bothersome.

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks.
IMHO only last is remove `[dependencies.glib]` and `[dependencies.gobject-sys]` in sys.

**Comment by GuillaumeGomez** [comment]
I'll check if nothing needs them in `cairo-sys` and if not, I'll remove them.

File: src/context.rs [hunk 1]
```diff
@@ -10,12 +10,17 @@ use std::ops;
 use ::paths::Path;
 use ::font::{TextExtents, TextCluster, FontExtents, ScaledFont, FontOptions, FontFace, Glyph};
 use ::matrices::{Matrix, MatrixTrait};
-use ffi::enums::{
+use ::enums::{
```
Comment (sdroege) [inline src/context.rs]: Or directly use the re-exported enums from the crate root?

File: src/lib.rs [hunk 2]
```diff
@@ -88,6 +87,8 @@ pub use enums::{
     RegionOverlap,
     SurfaceType,
 };
+#[cfg(any(feature = "v1_12", feature = "dox"))]
+pub use enums::MeshCorner;
```
Comment (sdroege) [inline src/lib.rs]: Or just `pub use enums::*;`?

**Comment by sdroege** [comment]
>  IMHO only last is remove `[dependencies.glib]` and `[dependencies.gobject-sys]` in sys.

`glib-sys` is still needed for all the `get_type()` functions in the -sys crate at least. `gobject-sys` might be unneeded

**Pull request merged**